core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -905,6 +910,23 @@ public void deleteTenant(int tenantId, boolean removeFromPersistentStorage)
         }
     }
 
+    /**
+     * Delete all tenant information related to tenant stored in UM tables.
+     *
+     * @param tenantId Id of the tenant
+     * @throws UserStoreException
+     */
+    public void deleteTenantUMData(int tenantId) throws UserStoreException {
```

Comment:
What is the default behavior of executing this? Are we deleting users and roles with the tenant deletion by default or do we have to enable the config?

Reply:
We have to enable this in config. It supposed to call from `carbon-multitenancy`. I think I missed to update the config template here. will add it.

Reply:
That configuration should honor only for Users and Roles.  Other information such as PERMISSIONS, Internal roles, Internal role mappings should be deleted irrespective of that config on tenant deletion.